src/Analyser/NodeScopeResolver.php

```diff
@@ -1909,6 +1909,17 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			$throwPoints = [];
 			if ($expr->name instanceof Expr) {
 				$nameType = $scope->getType($expr->name);
+				if (
+					$nameType->isObject()->yes()
+					&& $nameType->accepts(new ObjectType(Closure::class), true)->no()
+				) {
+					return $this->processExprNode(
+						new MethodCall($expr->name, '__invoke', $expr->getArgs(), $expr->getAttributes()),
```

Comment:
Hi, I don't love this patch. It's gonna change how `$foo()` is being processed and it's gonna add another error message. Besides CallCallablesRule it's also gonna invoke CallMethodsRule and that's not great.

Also I'm not sure `$nameType->accepts(new ObjectType(Closure::class), true)->no()` is trying to do and it doesn't seem like there's a test for this scenario.

I'd accept if we called `processExprNode` with a "noop" `$nodeCallback` and just taken `$throwPoints` from that.

Reply:
Tests are failing (https://github.com/phpstan/phpstan-src/actions/runs/6352469876/job/17255408049?pr=2502) without the extra check https://github.com/phpstan/phpstan-src/pull/2502/commits/838b9916f9350dd0b842c8dbe85716c38f047dab

This is because they are no method `__invoke` called on Closure